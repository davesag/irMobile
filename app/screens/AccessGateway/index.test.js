/**
 * @format
 */

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import AccessGatewayScreen, { mapStateToProps } from '.'

const apiKey = 'some-key'
const apiSecret = 'some-secret'

const requireAuth = true
const navigate = jest.fn()
const navigation = { navigate, getParam: jest.fn(() => 'blue') }

const resetStubs = () => {
  navigate.clearMock()
}

describe('mapStateToProps', () => {
  describe('when there are keys', () => {
    const stateProps = mapStateToProps({
      ir: { apiKey, apiSecret, busy: false }
    })

    it('returned hasKeys', () => {
      expect(stateProps).toHaveProperty('hasKeys', true)
    })

    it('returned loading', () => {
      expect(stateProps).toHaveProperty('loading', false)
    })
  })

  describe('when there are missing keys', () => {
    const stateProps = mapStateToProps({
      ir: { apiKey, apiSecret: null, requireAuth, busy: false }
    })

    it('returned hasKeys is false', () => {
      expect(stateProps).toHaveProperty('hasKeys', false)
    })

    it('returned loading', () => {
      expect(stateProps).toHaveProperty('loading', false)
    })
  })
})

describe('rendering', () => {
  let tree

  beforeAll(() => {
    tree = renderer.create(
      <AccessGatewayScreen loading navigation={navigation} />
    )
  })

  afterAll(resetStubs)

  it('rendered correctly', () => {
    expect(tree).toMatchSnapshot()
  })
})

describe('is loading', () => {
  beforeAll(() => {
    renderer
      .create(<AccessGatewayScreen loading={true} navigation={navigation} />)
      .getInstance()
      .componentDidUpdate()
  })

  afterAll(resetStubs)

  it("didn't call navigate", () => {
    expect(navigate).not.toHaveBeenCalled()
  })
})

describe('once loading is done', () => {
  describe('when keys arrive', () => {
    beforeAll(() => {
      renderer
        .create(
          <AccessGatewayScreen
            loading={false}
            navigation={navigation}
            hasKeys={true}
          />
        )
        .getInstance()
        .componentDidUpdate()
    })

    afterAll(resetStubs)

    it('called navigate("LoggedInStackNavigator")', () => {
      expect(navigate).toHaveBeenCalledWith('LoggedInStackNavigator')
    })
  })

  describe('when keys do not arrive', () => {
    beforeAll(() => {
      renderer
        .create(<AccessGatewayScreen loading={false} navigation={navigation} />)
        .getInstance()
        .componentDidUpdate()
    })

    afterAll(resetStubs)

    it('called navigate("LoggedOutStackNavigator")', () => {
      expect(navigate).toHaveBeenCalledWith('LoggedOutStackNavigator')
    })
  })
})
