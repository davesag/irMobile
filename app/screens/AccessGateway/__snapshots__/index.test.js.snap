// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering rendered correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
